src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -135,35 +136,63 @@ final case class PurposeApiServiceImpl(
     val operationLabel = s"Updating Purpose $purposeId"
     logger.info(operationLabel)
 
+    def isDraft(purpose: PurposeManagementDependency.Purpose): Boolean = {
+      val states = purpose.versions.map(_.state)
+      states == Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)
```

Comment:
This returns true only if exists just one version in draft state. Is this the requirement?

Reply:
Well, just one version can be in draft, so the impl is not wrong imho

Reply:
> Well, just one version can be in draft, so the impl is not wrong imho

Nope.
`purpose.versions.map(_.state)` can contain more than one state and just one of them could be in draft.
So comparing with `Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)`, return true only if exists one version in draft, but in the case of `Seq(PurposeManagementDependency.PurposeVersionState.ACTIVE, PurposeManagementDependency.PurposeVersionState.DRAFT)` it doesn't work.

Reply:
What I'm trying to understand here is if using a `.exists` or `.contains` should be the right choice.

Reply:
After some interactions with the team, I can confirm that this solution is fine.